plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java

```diff
@@ -91,23 +99,76 @@ public static IMarker[] getMarkersFromEditorSelection(final IStructuredSelection
 				.map(FordiacQuickFixHandler::getMarker).toArray(IMarker[]::new);
 	}
 
+	public static boolean hasMarker(final EditPart editPart) {
+		if (FordiacAnnotationUtil
+				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
+			final var annotation = annotationModel.getAnnotations(editPart.getModel());
+			return !annotation.isEmpty();
+		}
+		return false;
+	}
+
 	public static IMarker getMarker(final EditPart editPart) {
 		if (FordiacAnnotationUtil
 				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
 			final var annotation = annotationModel.getAnnotations(editPart.getModel());
+
 			if (!annotation.isEmpty()
 					&& annotation.iterator().next() instanceof final GraphicalMarkerAnnotation markerAnnotation) {
 				return markerAnnotation.getMarker();
 			}
+			if (!annotation.isEmpty() && annotation.iterator()
+					.next() instanceof final GraphicalValidationAnnotation validationAnnotation) {
+				createTemporaryMarker(validationAnnotation, annotationModel.getResource());
+				final Optional<GraphicalMarkerAnnotation> markerAnnotation = annotationModel
+						.getAnnotations(editPart.getModel()).stream()
+						.filter(GraphicalMarkerAnnotation.class::isInstance).map(GraphicalMarkerAnnotation.class::cast)
+						.findFirst();
+
+				if (markerAnnotation.isPresent()) {
+					return markerAnnotation.get().getMarker();
+				}
+			}
 		}
 		return null;
 	}
 
-	private static IMarker[] getAllMarker(final IResource resource) {
+	private static void createTemporaryMarker(final GraphicalValidationAnnotation validationAnnotation,
+			final IResource resource) {
 		try {
-			return resource.findMarkers(IMarker.MARKER, true, IResource.DEPTH_INFINITE);
-		} catch (final CoreException e) {
-			return new IMarker[0];
+			new WorkspaceModifyOperation() {
```

Comment:
Please only lock the current file by passing it to the `WorkspaceModifyOperation(ISchedulingRule)` constructor.